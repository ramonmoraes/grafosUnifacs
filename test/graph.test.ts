import Graph, { GraphLinks } from "../src/graph";
import GraphNode from "../src/graphNode";


describe("Graph", () => {
  let graph: Graph;
  beforeEach(() => {
    const nodes: GraphNode[] = [
      new GraphNode({
        identifier: "dart",
        attributes: { lang: ["en"], }
      }),
      new GraphNode({
        identifier: "ismelo",
        attributes: { lang: ["en"]}
      })
    ];
    graph = new Graph(nodes);
  });

  test("addNode", () => {
    expect(graph.nodes.length).toEqual(2);
    graph.addNode(new GraphNode({ identifier: "foo" }));
    expect(graph.nodes.length).toEqual(3);
  });

  test("removeNodeByIdentifier", () => {
    expect(graph.nodes.length).toEqual(2);
    graph.removeNodeByIdentifier("ismelo");
    expect(graph.nodes.length).toEqual(1);
  });

  test.only("getLinksBetweenNodes", () => {
    const links = graph.getLinkBetweenNodes(
      graph.nodes[0],
      graph.nodes[1]
    );

    expect(links.length).toEqual(1);
    expect(links[0].identifier).toEqual('lang');
    expect(links[0].value).toEqual('en');
    expect(links[0].connections[0]).toEqual(graph.nodes[0])
    expect(links[0].connections[1]).toEqual(graph.nodes[1])
  });
});
