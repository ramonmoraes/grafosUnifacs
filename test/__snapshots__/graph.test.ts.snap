// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graph getAdjacentNodesByIdentifier 1`] = `
Object {
  "max": 2,
  "med": 1.3333333333333333,
  "min": 1,
}
`;

exports[`Graph getLinksByIdentifier 1`] = `
Array [
  Object {
    "connections": Array [
      GraphNode {
        "attributes": Object {
          "lang": Array [
            "en",
            "pt",
          ],
        },
        "identifier": "ismelo",
        "removeAttribute": [Function],
      },
      GraphNode {
        "attributes": Object {
          "lang": Array [
            "en",
          ],
        },
        "identifier": "dart",
        "removeAttribute": [Function],
      },
    ],
    "identifier": "lang",
    "value": "en",
  },
]
`;

exports[`Graph getLinksByIdentifier 2`] = `
Array [
  Object {
    "connections": Array [
      GraphNode {
        "attributes": Object {
          "lang": Array [
            "en",
            "pt",
          ],
        },
        "identifier": "ismelo",
        "removeAttribute": [Function],
      },
      GraphNode {
        "attributes": Object {
          "lang": Array [
            "en",
          ],
        },
        "identifier": "dart",
        "removeAttribute": [Function],
      },
    ],
    "identifier": "lang",
    "value": "en",
  },
  Object {
    "connections": Array [
      GraphNode {
        "attributes": Object {
          "lang": Array [
            "en",
            "pt",
          ],
        },
        "identifier": "ismelo",
        "removeAttribute": [Function],
      },
      GraphNode {
        "attributes": Object {
          "lang": Array [
            "pt",
          ],
        },
        "identifier": "foo",
        "removeAttribute": [Function],
      },
    ],
    "identifier": "lang",
    "value": "pt",
  },
]
`;
